OpenRA.Mods.Common/Widgets/WorldCommandWidget.cs

```diff
@@ -191,15 +191,24 @@ bool CycleBases()
 				.Select(b => b.Actor)
 				.ToList();
 
-			if (!bases.Any())
-				return true;
+            // If no base is found all existing buildings are selected.
+            if (!bases.Any())
+            {
+				bases = world.ActorsWithTrait<TargetableBuilding>()
+                .Where(a => a.Actor.Owner == world.LocalPlayer)
+                .Select(b => b.Actor)
+                .ToList();
+                if (!bases.Any())
+					return true;
+            }   
```

Comment:
> OpenRA.Mods.Common/Widgets/WorldCommandWidget.cs:L203: [SP2000] Invalid spacing at the end of the line.

You should run make check to notice issues like having additional whitespace at the end of a line.